### DISCUSSION TITLE
Browser support

### DISCUSSION THREAD

**Discussion by aurelien-boubennec** [W4G1/multithreading#4] [General]
Hello,

Does it really work in the browser or is it just for Node?
I am asking this because I'm getting this error message:

> ERROR in node:worker_threads
> Module build failed: UnhandledSchemeError: Reading from "node:worker_threads" is not handled by plugins (Unhandled scheme).
> Webpack supports "data:" and "file:" URIs by default.
> You may need an additional plugin to handle "node:" URIs.

**Comment by W4G1** [comment]
Hi! Yes it should work in the browser. What framework or other tooling are you using?

**Reply by aurelien-boubennec** [reply]
Hey, wow that was fast! I'm using React 18 with webpack and typescript. My app was built with the Create React App.

**Reply by W4G1** [reply]
Interesting. I have tested with React using Vite, but not with webpack, so I think the webpack bundling is causing this issue. Unfortunately I don't have a lot of spare time on my hands to investigate, but I'll see what I can do.

In the meantime, you might have more success using React with Vite.

**Reply by aurelien-boubennec** [reply]
No problem, I guess I need to override the webpack config and add the appropriate plugin.
Thanks!

**Reply by aurelien-boubennec** [reply]
I migrated a small project to Vite and had your library working without trouble.
I have to say, so far I'm impressed!
Is there a way to debug the function that is in a separate thread?

**Reply by W4G1** [reply]
I haven't tested with any debugging tools aside from the notorious `console.log()` 😄

However, debugging errors with multithreading is a lot nicer because it automatically injects enhanced stacktraces to any error that is thrown inside a thread, which includes the thread number and line where the error has occurred:

![Example of an enhanced stack trace](https://github.com/W4G1/multithreading/assets/38042257/4c5b3352-ad19-4270-86e4-dad6fc6e0fe6)
